nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/DynamicCustomOp.java

```diff
@@ -58,8 +58,53 @@ public DynamicCustomOp() {
         tArguments = new ArrayList<>();
     }
 
+
+    // Hack to get output shape without data
+    private Map<long[], INDArray> dummyArrays = new HashMap<>();
```

Comment:
At this moment DCO isn't a long living object, so each instance of any op will have own map. What's the idea here?

Reply:
The idea is we pass an INDArray (zeros) of same shape as inputs. In case we have already created an INDArray for a given shape, we re use it.

Reply:
What is the larger context here? We can already auto infer op shapes now in the op executioner.

Reply:
But it fails for samediff ops which do not have calculateOutputShape() defined. SDVariable.getShape() returns[].